src/box/sql/build.c

```diff
@@ -3389,3 +3389,55 @@ sql_setting_set(struct Parse *parse_context, struct Token *name,
 	sqlVdbeAddOp4(vdbe, OP_SetSession, target, 0, 0, key, P4_DYNAMIC);
 	return;
 }
+
+void
+sql_emit_show_create_table_one(struct Parse *parse, struct Token *name)
+{
+	struct Vdbe *v = sqlGetVdbe(parse);
+	char *space_name = sql_name_from_token(name);
+	sqlVdbeSetNumCols(v, 1);
+	vdbe_metadata_set_col_name(v, 0, "CREATE TABLE statement");
+	vdbe_metadata_set_col_type(v, 0, "string");
+
+	int cursor = parse->nTab++;
+	int space_reg = ++parse->nMem;
+	int name_reg = ++parse->nMem;
+	sqlVdbeAddOp2(v, OP_OpenSpace, space_reg, BOX_VSPACE_ID);
+	sqlVdbeAddOp3(v, OP_IteratorOpen, cursor, 2, space_reg);
+	sqlVdbeChangeP5(v, OPFLAG_SEEKEQ);
+	sqlVdbeAddOp4(v, OP_String8, 0, name_reg, 0, space_name, P4_DYNAMIC);
+	int addr1 = sqlVdbeAddOp4Int(v, OP_SeekGE, cursor, 0, name_reg, 1);
+	int addr2 = sqlVdbeAddOp4Int(v, OP_IdxGT, cursor, 0, name_reg, 1);
+	int space_id_reg = ++parse->nMem;
+	sqlVdbeAddOp3(v, OP_Column, cursor, BOX_SPACE_FIELD_ID, space_id_reg);
+	int result_reg = ++parse->nMem;
+	sqlVdbeAddOp2(v, OP_ShowTable, space_id_reg, result_reg);
+	sqlVdbeAddOp2(v, OP_ResultRow, result_reg, 1);
+	sqlVdbeJumpHere(v, addr1);
+	sqlVdbeJumpHere(v, addr2);
+}
+
+void
+sql_emit_show_create_table_all(struct Parse *parse)
+{
+	struct Vdbe *v = sqlGetVdbe(parse);
+	sqlVdbeSetNumCols(v, 1);
+	vdbe_metadata_set_col_name(v, 0, "CREATE TABLE statement");
+	vdbe_metadata_set_col_type(v, 0, "string");
+
+	int cursor = parse->nTab++;
+	int space_reg = ++parse->nMem;
+	int key_reg = ++parse->nMem;
+	sqlVdbeAddOp2(v, OP_OpenSpace, space_reg, BOX_VSPACE_ID);
+	sqlVdbeAddOp3(v, OP_IteratorOpen, cursor, 0, space_reg);
+	sqlVdbeAddOp2(v, OP_Integer, BOX_SYSTEM_ID_MAX, key_reg);
+	int addr1 = sqlVdbeAddOp4Int(v, OP_SeekGT, cursor, 0, key_reg, 1);
+	int space_id_reg = ++parse->nMem;
+	int addr2 = sqlVdbeAddOp3(v, OP_Column, cursor, BOX_SPACE_FIELD_ID,
+				  space_id_reg);
+	int result_reg = ++parse->nMem;
+	sqlVdbeAddOp2(v, OP_ShowTable, space_id_reg, result_reg);
+	sqlVdbeAddOp2(v, OP_ResultRow, result_reg, 1);
+	sqlVdbeAddOp2(v, OP_Next, cursor, addr2);
+	sqlVdbeJumpHere(v, addr1);
```

Comment:
Why do you need so many additional opcodes if you anyway added `OP_ShowTable`? Couldn't space name simply be its direct argument?

Reply:
Ah, I see, it is called for each tuple during iteration. Not once per vdbe exec.